File: bench/langs/ledger.ml
(* Keeps the running balance for one account. *)

let limit = 5000

let refuse owner amount =
  warn_owner owner;
  0

let commit_entry owner amount =
  write_entry owner amount;
  amount

(* Bills the account and returns what is left. *)
let charge owner amount =
  if amount > limit then
    refuse owner amount
  else
    commit_entry owner amount

File: bench/langs/ledger.py
"""Keeps the running balance for one account."""

LIMIT = 5000


def refuse(owner, amount):
    """Rejects a charge over the limit."""
    warn_owner(owner)
    return 0


def commit_entry(owner, amount):
    write_entry(owner, amount)
    return amount


def charge(owner, amount):
    """Bills the account and returns what is left."""
    if amount > LIMIT:
        return refuse(owner, amount)
    return commit_entry(owner, amount)
